saddle-core/src/main/scala/org/saddle/array/package.scala

```diff
@@ -43,6 +43,26 @@ package object array {
     }
   }
 
+  /** Copies a contiguous segment of the array into a new array.
+    *
+    * @param from
+    *   start index, inclusive
+    * @param until
+    *   end index, exclusive
+    * @param source
+    *   source array to copy
+    * @return
+    */
+  def copySlice[@spec(Boolean, Int, Long, Double) T: ST](
+      from: Int,
+      until: Int,
+      source: Array[T]
+  ): Array[T] = {
+    val target = empty[T](until - from)
+    System.arraycopy(source, from, target, 0, until - from)
```

Comment:
There seems to be a more convenient method `Arrays.copyOfRange` for this: https://stackoverflow.com/a/4439612/4296401

Reply:
Not sure how it plays with specialization though.

Reply:
Thanks, indeed.

copyOfRange would need to be put into ScalarTag. That breaks binary compatibility because ScalarTag is technically open for extension by client code (but it should not be). 

Probably I would prefer copyOfRange on the primitive types because it is marked as jvm intrinsic. 

I merge this now and will come back to copyOfRange and the scalartag binary compatibility.

Reply:
> That breaks binary compatibility because ScalarTag is technically open for extension by client code (but it should not be).

I wouldn't worry about that breaking compat on this. I don't see why users would extend it.